src/var.c

```diff
@@ -353,18 +354,37 @@ int DEFAULT_SYMVER_PRE(fi_param_get)(struct fi_provider *provider,
 	if (!param)
 		return -FI_ENOENT;
 
-	str_value = getenv(param->env_var_name);
-	if (!str_value) {
-		FI_DBG(provider, FI_LOG_CORE,
-			"variable %s=<not set>, checking system\n", param_name);
+	if (ofi_prefer_sysconfig) {
 		conf = find_conf_entry(param->env_var_name);
 		if (!conf) {
-			FI_INFO(provider, FI_LOG_CORE,
-				"variable %s=<not set>\n", param_name);
-			ret = -FI_ENODATA;
-			goto out;
+			FI_DBG(provider, FI_LOG_CORE,
+			       "variable %s=<not set>, checking environment\n",
+			       param_name);
+			str_value = getenv(param->env_var_name);
+			if (!str_value) {
+				FI_INFO(provider, FI_LOG_CORE,
+					"variable %s=<not set>\n", param_name);
+				ret = -FI_ENODATA;
+				goto out;
+			}
+		} else {
+			str_value = conf->value;
+		}
+	} else {
+		str_value = getenv(param->env_var_name);
+		if (!str_value) {
+			FI_DBG(provider, FI_LOG_CORE,
+			       "variable %s=<not set>, checking system\n",
+			       param_name);
+			conf = find_conf_entry(param->env_var_name);
+			if (!conf) {
+				FI_INFO(provider, FI_LOG_CORE,
+					"variable %s=<not set>\n", param_name);
+				ret = -FI_ENODATA;
+				goto out;
+			}
+			str_value = conf->value;
 		}
-		str_value = conf->value;
 	}
```

Comment:
To reduce code duplication, you can read both conf and env first, and choose the right one based on preference. Something like:
```
if (ofi_prefer_sysconfig)
    str_value = conf->value ? conf->value : env_value;
else
    str_value = env_value ? env_value : conf->value;

if (!str_value)
    FI_DBG(.....);
```

Reply:
I did think about this and can move to this if preferred. my concern was adding extra calls to query the environment or looping through the config cache.  I suppose the expectation is the cache should be pretty small so querying the environment would be the bigger concern, but we're doing that anyway today so maybe my concern was misplaced...

Reply:
updated to remove duplication!